### PR TITLE
RPC formatting upgrade

### PR DESCRIPTION
**Pull request by flicker-harmony**
## Issue

First part of rpc formatting upgrade in https://github.com/harmony-one/harmony/issues/2033. 

## Test
 
Tested locally. But needs to be tested by others, on testnet, by p-ops team possibly with @Daniel-VDM help.
Screenshot 2020-01-07 at 23 45 24
Screenshot 2020-01-07 at 23 44 56

### Unit Test Coverage

Before:

```
<!-- copy/paste 'go test -cover' result in the directory you made change -->
```

After:

```
<!-- copy/paste 'go test -cover' result in the directory you made change -->
```

## Operational Checklist

1. **Does this PR introduce backward-incompatible changes to the on-disk data structure and/or the over-the-wire protocol?**. (If no, skip to question 8.)

    NO

2. **Describe the migration plan.**. For each flag epoch, describe what changes take place at the flag epoch, the anticipated interactions between upgraded/non-upgraded nodes, and any special operational considerations for the migration.

3. **Describe how the plan was tested.**

4. **How much minimum baking period after the last flag epoch should we allow on Pangaea before promotion onto mainnet?**

5. **What are the planned flag epoch numbers and their ETAs on Pangaea?**

6. **What are the planned flag epoch numbers and their ETAs on mainnet?**

    Note that this must be enough to cover baking period on Pangaea.

7. **What should node operators know about this planned change?**

8. **Does this PR introduce backward-incompatible changes *NOT* related to on-disk data structure and/or over-the-wire protocol?** (If no, continue to question 11.)

    NO

9. **Does the existing `node.sh` continue to work with this change?**

10. **What should node operators know about this change?**

    Will need to adjust rpc format in external apps like explorers, sdk, etc.

11. **Does this PR introduce significant changes to the operational requirements of the node software, such as >20% increase in CPU, memory, and/or disk usage?**

   NO

## TODO

1. Testing: on testnet, by p-ops team, by partners, with @Daniel-VDM  help. Test all rpc methods one by one.
2. SDK, explorers backends adjustments, notify partners and node operators to adjust accordingly, update wiki (just replace hex to decimal everywhere and examples

### CODE REVIEW

**Comment by LeoHChen** [comment]
As requested by TrustWallet for backward compatibility and based on my understanding, we may have to use apiv2 to support the new format of decimal values.  Please check with Victor from trustwallet team.  I am afraid this change will break their wallet.  Also, how about our own explorer?

**Comment by flicker-harmony** [comment]
@LeoHChen it will break lots of apps and I noted it in issue and this PR, there is no other way except making it as v2 apis.

**Comment by LeoHChen** [comment]
then we need to make it an api v2.

**Comment by flicker-harmony** [comment]
We will make separate folders v1 v2 and apis with prefixes to support old apis.

File: internal/hmyapi/blockchain.go [hunk 1]
```diff
@@ -300,19 +299,19 @@ func (s *PublicBlockChainAPI) GetLeader(ctx context.Context) string {
 }
 
 // GetValidatorSelfDelegation returns validator stake.
-func (s *PublicBlockChainAPI) GetValidatorSelfDelegation(ctx context.Context, address string) hexutil.Uint64 {
-	return hexutil.Uint64(s.b.GetValidatorSelfDelegation(internal_common.ParseAddr(address)).Uint64())
+func (s *PublicBlockChainAPI) GetValidatorSelfDelegation(ctx context.Context, address string) uint64 {
```
Comment (rlan35) [inline internal/hmyapi/blockchain.go]: For all token related value, we should use big.Int. uint64 is not enough for represent token more than 18 ONE. which is 18000000000000000000

**Comment by neeboo** [comment]
Thats breaking changes.

**Comment by flicker-harmony** [comment]
@neeboo it supports new and old one now. Developers can adjust whenever they see fit and we can make smooth updates for sdk, explorers, apps. Of course at some point we will stop updating v1 except issues.

**Pull request merged**